ckan/logic/validators.py

```diff
@@ -562,6 +562,59 @@ def ignore_not_sysadmin(key: FlattenKey, data: FlattenDataDict,
     data.pop(key)
 
 
+def limit_sysadmin_update(key: FlattenKey, data: FlattenDataDict,
+                          errors: FlattenErrorDict, context: Context) -> Any:
+    """
+    Should not be able to modify your own sysadmin privs, or the system user's
+    """
+    value = data.get(key)
+
+    if value is None:
+        # sysadmin key not in data, return here
+        data.pop(key, None)
+        return
+
+    contextual_user = context.get('auth_user_obj')
+    site_id = config.get('ckan.site_id')
+    contextual_user_name = context.get('user')
+
+    if not contextual_user:
+        # auth_user_obj has not been set, try to get it from user context
+        contextual_user = context['model'].User.get(contextual_user_name)
+
+        if contextual_user:
+            contextual_user_name = contextual_user.name
+        elif not context.get('ignore_auth'):
+            # using a fake user and not ignoring auth, silently fail here
+            data.pop(key)
+            return
```

Comment:
This whole block can probably be simplified to this right?
I don't understand the `ignore_auth` bit. This validator should be not related to authorization, unless I'm missing something

```suggestion
    if not contextual_user and contextual_user_name:
        contextual_user = context['model'].User.get(contextual_user_name)

    if not contextual_user:
        return
```

Reply:
@amercader yeee it was kind of a bad workaround to this piece of code in the Reset Password view: https://github.com/ckan/ckan/blob/master/ckan/views/user.py#L809

That piece of code in the reset password view could probably be moved into a validator for the user schema??